src/components/legend/draw.js

```diff
@@ -329,9 +329,31 @@ module.exports = function draw(gd) {
                 xf = dragElement.align(newX, 0, gs.l, gs.l + gs.w, opts.xanchor);
                 yf = dragElement.align(newY, 0, gs.t + gs.h, gs.t, opts.yanchor);
             },
-            doneFn: function(dragged) {
+            doneFn: function(dragged, numClicks, e) {
                 if(dragged && xf !== undefined && yf !== undefined) {
                     Plotly.relayout(gd, {'legend.x': xf, 'legend.y': yf});
+                } else {
+                    var traces = [],
+                        clickedTrace;
+                    fullLayout._infolayer.selectAll('g.traces').each(function() { d3.select(this).call(function() { traces.push(this); }); });
+                    for(var i = 0; i < traces.length; i++) {
+                        var tracei = traces[i];
+                        var p = tracei[0][0].getBoundingClientRect();
+                        if(e.clientX >= p.left && e.clientX <= p.right && e.clientY >= p.top && e.clientY <= p.bottom) {
+                            clickedTrace = tracei;
+                            break;
+                        }
+                    }
```

Comment:
This is a funny pattern... the two `this`'s got me totally confused at first. Can't you just do something like:
```js
clickedTrace = fullLayout._infolayer.selectAll('g.traces').filter(function() {
    var bbox = this.getBoundingClientRect();
    return (e.clientX >= bbox.left && e.clientX <= bbox.right && 
        e.clientY >= bbox.top && e.clientY <= bbox.bottom);
});
if(clickedTrace.size()) {
    ...
```